src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -139,6 +146,28 @@ impl Window {
             )
             .map_err(|_| os_error!(OsError::WaylandMisc("failed to create window.")))?;
 
+        // Set CSD frame config
+        #[cfg(feature = "sctk-adwaita")]
+        if let Some(theme) = platform_attributes.csd_theme {
+            let config = match theme {
+                Theme::Light => sctk_adwaita::FrameConfig::light(),
+                Theme::Dark => sctk_adwaita::FrameConfig::dark(),
+            };
+            window.set_frame_config(config);
+        } else {
+            if let Ok(env) = std::env::var(WAYLAND_CSD_THEME_ENV_VAR) {
+                match env.to_lowercase().as_str() {
+                    "dark" => {
+                        window.set_frame_config(sctk_adwaita::FrameConfig::dark());
+                    }
+                    "light" => {
+                        window.set_frame_config(sctk_adwaita::FrameConfig::light());
+                    }
+                    _ => {}
+                }
+            }
+        }
```

Comment:
Could you refactor all of that into something like

```rust
 #[cfg(feature = "sctk-adwaita")]
{
    let theme = platform_attributes.csd_theme.unwrap_or_else(|| {
        match std::env::var(WAYLAND_CSD_THEME_ENV_VAR).unwrap_or_default("") {
            "dark" => Theme::Dark,
            _ => Theme::Light
         }
    })
    let config = match theme {
         Theme::Light => sctk_adwaita::FrameConfig::light(),
         Theme::Dark => sctk_adwaita::FrameConfig::dark(),
    };
    window.set_frame_cofig(config)
}
```

I think it would be shorter and more readable.